core/state/intra_block_state.go

```diff
@@ -827,3 +828,50 @@ func (sdb *IntraBlockState) AddressInAccessList(addr libcommon.Address) bool {
 func (sdb *IntraBlockState) SlotInAccessList(addr libcommon.Address, slot libcommon.Hash) (addressPresent bool, slotPresent bool) {
 	return sdb.accessList.Contains(addr, slot)
 }
+
+// ValidateKnownAccounts validates the knownAccounts passed in the options parameter in the conditional transaction (EIP-4337)
+func (sdb *IntraBlockState) ValidateKnownAccounts(knownAccounts types2.KnownAccountStorageConditions) error {
+	if knownAccounts == nil {
+		return nil
+	}
+
+	for address, condition := range knownAccounts {
+		tempAccount, err := sdb.stateReader.ReadAccountData(address)
+		if err != nil {
+			return fmt.Errorf("error reading account data at: %v", address)
+		}
+
+		// check if the value is hex string or an object
+		switch {
```

Comment:
looks like now we could move the `tempAccount` check and `else` logic outside switch before it, is it ok for you?

```
if tempAccount == nil {
    return fmt.Errorf("Storage Trie is nil for: %v", address)
}

// check if the value is hex string or an object
switch {
...
```

Reply:
[Done](https://github.com/ledgerwatch/erigon/pull/8229/commits/a6a66ad1435a154dead5a168ff081b3f2b489fa2).